pkg/option/resolver/resolver.go

```diff
@@ -89,7 +97,18 @@ func ResolveConfigurations(ctx context.Context, client client.Clientset, nodeNam
 		}
 	}
 
-	config["config-sources"] = strings.Join(sourceDescriptions, ",")
+	sConfigJson, err := json.Marshal(sourceDescriptions)
+	if err != nil {
+		return config, fmt.Errorf("encoding to JSON %s: %w", ConfigSources, err)
+	}
+
+	oConfigJson, err := json.Marshal(ConfigOverride{AllowConfigKeys: allowConfigKeys, DenyConfigKeys: denyConfigKeys})
+	if err != nil {
+		return config, fmt.Errorf("encoding to JSON %s: %w", ConfigSourcesOverrides, err)
```

Comment:
This error msg should at least be disambiguated from the error msg above in order to keep it easier for us to know where the error came from

Reply:
we have the additional `ConfigSources` / `ConfigSourcesOverrides` log key which allows us to distinguish the msg source.